### PR TITLE
[FEATURE] control over drawing of composer table grid horizontal & vertical lines

### PR DESCRIPTION
**Pull request by nirvn**
This PR expands the composer table's grid styling possibilities by adding a pair of settings to enable/disable rendering of the grid's horizontal and vertical lines, which allows users to end up with nice-looking tables like this:
untitled

Or:
untitled

@nyalldawson , I've (finally) remember to add tests this time around ;) the PR should fail first as it'll need mask adjustment.

### CODE REVIEW

File: src/core/composer/qgscomposertablev2.cpp [hunk 1]
```diff
@@ -534,15 +541,15 @@ void QgsComposerTableV2::render( QPainter *p, const QRectF &, const int frameInd
     gridPen.setColor( mGridColor );
     gridPen.setJoinStyle( Qt::MiterJoin );
     p->setPen( gridPen );
-    drawHorizontalGridLines( p, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader );
-    drawVerticalGridLines( p, mMaxColumnWidthMap, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader, mergeCells );
+    if ( mHorizontalGrid ) drawHorizontalGridLines( p, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader );
```
Comment (nyalldawson) [inline src/core/composer/qgscomposertablev2.cpp]: Pretty sure everyone hates single line if statements :)

File: src/core/composer/qgscomposertablev2.h [hunk 2]
```diff
@@ -338,6 +338,42 @@ class CORE_EXPORT QgsComposerTableV2: public QgsComposerMultiFrame
      */
     QColor gridColor() const { return mGridColor; }
 
+    /** Sets whether the grid's horizontal lines should be drawn in the table
+     * @param horizontalGird set to true to draw grid's horizontal lines
+     * @see setShowGrid
+     * @see setGridStrokeWidth
+     * @see setGridColor
+     * @see setVerticalGrid
```
Comment (nyalldawson) [inline src/core/composer/qgscomposertablev2.h]: Can you add some  `@note added in QGIS 3.0` s?

File: tests/src/core/testqgscomposertablev2.cpp [hunk 3]
```diff
@@ -675,6 +677,80 @@ void TestQgsComposerTableV2::multiLineText()
...
+  f3.setAttribute( "col1", "singleline" );
+  f3.setAttribute( "col2", "singleline" );
+  f3.setAttribute( "col3", "multiline\nstring" );
+  QgsFeature f4( multiLineLayer->dataProvider()->fields(), 4 );
+  f4.setAttribute( "col1", "long triple\nline\nstring" );
+  f4.setAttribute( "col2", "double\nlinestring" );
+  f4.setAttribute( "col3", "singleline" );
+  multiLineLayer->dataProvider()->addFeatures( QgsFeatureList() << f1 << f2 << f3 << f4 );
+
+  mFrame2->setSceneRect( QRectF( 5, 40, 100, 90 ) );
```
Comment (nyalldawson) [inline tests/src/core/testqgscomposertablev2.cpp]: Can you turn off the item border for these frames in these two tests? It'll help the test ensure that the top/left/right lines aren't drawn when not desired.

**Comment by nyalldawson** [comment]
Looks great!

File: src/core/composer/qgscomposertablev2.cpp (hunk 1 above)
Comment (nirvn) [inline src/core/composer/qgscomposertablev2.cpp]: @nyalldawson , darn, I was sure I'd found the one scenario where single line if statement would be embraced ... ;)

**Comment by nirvn** [comment]
@nyalldawson , thanks for the quick review, it's appreciated.

All of your comments were addressed in the updated commit. Waiting for the PR to give a rendered image to the newly-added test to create a mask.

**Comment by nirvn** [comment]
@nyalldawson , thanks for being so quick at reviewing, and always supportive.

**Pull request merged**